File: src/scene_bitmap_test.cpp
#include "scene_bitmap_test.h"

#include "input.h"
#include "sprite.h"
#include "bitmap.h"
#include "player.h"
#include "output.h"
#include "font.h"
#include "graphics.h"

#include <boost/assert.hpp>


Scene_BitmapTest::Scene_BitmapTest()
		: Scene("Bitmap Test")
		, rgba_(new Sprite())
		, text_(new Sprite())
{
	rgba_->SetBitmap(Bitmap::Create(40, 10));
	rgba_->SetY(0);

	assert(rgba_->GetBitmap()->get_pixel(0, 0) == Color(0, 0, 0, 0));

	rgba_->GetBitmap()->fill(Rect( 0, 0, 10, 10), Color(255, 0, 0, 255));
	rgba_->GetBitmap()->fill(Rect(10, 0, 10, 10), Color(0, 255, 0, 255));
	rgba_->GetBitmap()->fill(Rect(20, 0, 10, 10), Color(0, 0, 255, 255));
	rgba_->GetBitmap()->fill(Rect(30, 0, 10, 10), Color(255, 255, 255, 0));

	assert(rgba_->GetBitmap()->get_pixel( 0, 0) == Color(255, 0, 0, 255));
	assert(rgba_->GetBitmap()->get_pixel(10, 0) == Color(0, 255, 0, 255));
	assert(rgba_->GetBitmap()->get_pixel(20, 0) == Color(0, 0, 255, 255));
	assert(rgba_->GetBitmap()->get_pixel(30, 0) == Color(255, 255, 255, 0));

	Font::default_color = Color(255, 255, 255, 255);
	text_->SetY(10);
	text_->SetBitmap(Bitmap::Create(100, 12));
	text_->GetBitmap()->fill(Rect(0, 0, 100, 12), Color(0, 0, 255, 255));
	text_->GetBitmap()->draw_text(0, 0, "Hello World!");
}

void Scene_BitmapTest::Update() {
	if(Input().IsAnyPressed()) {
		BOOST_VERIFY(Output().TakeScreenshot());
		Player().exit_flag = true;
	}
}

File: mrblib/scene_bitmap_test.rb
class Scene_BitmapTest < Scene
	def initialize
    super 'Bitmap Test'

    @rgba = Sprite.new
    @rgba.bitmap = Bitmap.new 40, 10
    @rgba.y = 0
    assert @rgba.bitmap.get_pixe(0, 0) == Color.new(0, 0, 0, 0)

    @rgba.bitmap.fill Rect.new( 0, 0, 10, 10), Color.new(255, 0, 0, 255)
    @rgba.bitmap.fill Rect.new(10, 0, 10, 10), Color.new(0, 255, 0, 255)
    @rgba.bitmap.fill Rect.new(20, 0, 10, 10), Color.new(0, 0, 255, 255)
    @rgba.bitmap.fill Rect.new(30, 0, 10, 10), Color.new(255, 255, 255, 255)
    assert @rgba.bitmap.get_pixel( 0, 0) == Color.new(255, 0, 0, 255)
    assert @rgba.bitmap.get_pixel(10, 0) == Color.new(0, 255, 0, 255)
    assert @rgba.bitmap.get_pixel(20, 0) == Color.new(0, 0, 255, 255)
    assert @rgba.bitmap.get_pixel(30, 0) == Color.new(255, 255, 255, 255)

    @text = Sprite.new
    Font.default_color = Color.new 255, 255, 255, 255
    @text.y = 10
    @text.bitmap = Bitmap.new 100, 12
    @text.bitmap.fill Rect.new(0, 0, 100, 12), Color.new(0, 0, 255, 255)
    @text.bitmap.draw_text_2k 0, 0, "Hello World!"
  end

  def update
    if Input.any_pressed?
      Output.take_screenshot
      Player.exit_flag = true
    end
  end
end
